stopes/ui/seamlisten/backend/app/fileviewer.py

```diff
@@ -131,7 +133,14 @@ def get_annotations(
 @router.post("/general/")
 async def general_query(query: DefaultQuery) -> Response:
     query_path = query.gz_path.strip()
-    print(query_path.split(" "))
+    # Convert the path to an absolute path
+    query_path = str(Path(os.path.expanduser(query_path)).resolve())
+    # Check if the path is a directory
+    print(query_path)
+    if os.path.isdir(query_path):
```

Comment:
just keep query_path as a Path instead of casting it back to  str and then you can use the Path functions to check isdir